app/Console/Commands/RebuildQueryserviceData.php

```diff
@@ -0,0 +1,153 @@
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use App\Wiki;
+use App\WikiSetting;
+use App\QueryserviceNamespace;
+use App\Jobs\SpawnQueryserviceUpdaterJob;
+
+class RebuildQueryserviceData extends Command
+{
+    private const NAMESPACE_ITEM = 120;
+    private const NAMESPACE_PROPERTY = 122;
+    private const NAMESPACE_LEXEME = 146;
+
+    protected $signature = 'wbs-qs:rebuild {--domain=*} {--chunkSize=50} {--sparqlUrlFormat=http://queryservice.default.svc.cluster.local:9999/bigdata/namespace/%s/sparql}';
+
+    protected $description = 'Rebuild the queryservice data for a certain wiki or all wikis';
+
+    protected int $chunkSize;
+    protected string $apiUrl;
+    protected string $sparqlUrlFormat;
+
+    public function handle()
+    {
+        $this->chunkSize = intval($this->option('chunkSize'));
+        $this->sparqlUrlFormat = $this->option('sparqlUrlFormat');
+        $this->apiUrl = getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php';
+
+        $wikiDomains = $this->option('domain');
+        $exitCode = 0;
+
+        $wikis = count($wikiDomains) !== 0
+            ? Wiki::whereIn('domain', $wikiDomains)->get()
+            : Wiki::query()->get();
+
+        foreach ($wikis as $wiki) {
+            try {
+                $entities = $this->getEntitiesForWiki($wiki);
+                $sparqlUrl = $this->getSparqlUrl($wiki);
+            } catch (\Exception $ex) {
+                Log::error(
+                    'Failed to get prerequisites for enqueuing wiki '.$wiki->domain.', will not dispatch jobs.'
+                );
+                Log::error($ex);
+                $exitCode = 1;
+                break;
+            }
+
+            $entityChunks = array_chunk($entities, $this->chunkSize);
+            foreach ($entityChunks as $entityChunk) {
+                dispatch(
+                    new SpawnQueryserviceUpdaterJob(
+                        $wiki->domain,
+                        implode(',', $entityChunk),
+                        $sparqlUrl,
+                    )
+                );
+            }
+        }
+
+        return $exitCode;
+    }
+
+    private function getEntitiesForWiki (Wiki $wiki): array
+    {
+        $items = $this->fetchPagesInNamespace($wiki->domain, self::NAMESPACE_ITEM);
+        $properties = $this->fetchPagesInNamespace($wiki->domain, self::NAMESPACE_PROPERTY);
+
+        $lexemesSetting = WikiSetting::where(
+            [
+                'wiki_id' => $wiki->id,
+                'name' => WikiSetting::wwExtEnableWikibaseLexeme
+            ],
+        )->first();
+        $hasLexemesEnabled = $lexemesSetting !== null && $lexemesSetting->value === '1';
+        $lexemes = $hasLexemesEnabled
+            ? $this->fetchPagesInNamespace($wiki->domain, self::NAMESPACE_LEXEME)
+            : [];
+
+        $merged = array_merge($items, $properties, $lexemes);
+        return $this->stripPrefixes($merged);
```

Comment:
I'm intrigued that this syntax works when the method is declared `static`; Until now I thought in that case you need to write `self::methodName()` or `className::methodName()`

Reply:
Are there any downsides to using `$this` here? Happy to change it to `self`, even if it's just for being more idiomatic.

Reply:
I couldn't find any real downsides, gotta love this SO answer, I wasn't aware the other way around is also possible sometimes apparently...
> This means that -> is mostly used to access instance members (though it can also be used to access static members, such usage is discouraged), while :: is usually used to access static members (though in a few special cases, it's used to access instance members).

https://stackoverflow.com/questions/3173501/whats-the-difference-between-double-colon-and-arrow-in-php